execution/executor-benchmark/src/fake_executor.rs

```diff
@@ -142,20 +106,88 @@ impl FakeExecutor {
                 sender_coin_store_key,
                 WriteOp::Modification(bcs::to_bytes(&sender_coin_store)?),
             ),
-            (
-                new_account_key,
-                WriteOp::Creation(bcs::to_bytes(&new_account)?),
-            ),
-            (
-                new_coin_store_key,
-                WriteOp::Creation(bcs::to_bytes(&new_coin_store)?),
-            ),
             (
                 TOTAL_SUPPLY_STATE_KEY.clone(),
                 WriteOp::Modification(bcs::to_bytes(&total_supply)?),
             ),
         ];
 
+        if new_account.is_some() {
+            if fail_on_existing {
+                return Ok(TransactionOutput::new(
+                    Default::default(),
+                    vec![],
+                    0,
+                    TransactionStatus::Keep(ExecutionStatus::MoveAbort {
+                        location: AbortLocation::Module(ModuleId::new(
+                            AccountAddress::ONE,
+                            Identifier::from_str("account").unwrap(),
+                        )),
+                        code: 7,
+                        info: None,
+                    }),
+                ));
+            }
+
+            let mut new_coin_store = {
+                let _timer = TIMER
+                    .with_label_values(&["read_new_coin_store"])
+                    .start_timer();
+                DbAccessUtil::get_coin_store(&new_coin_store_key, state_view)?.unwrap()
+            };
+
+            if initial_balance != 0 {
+                new_coin_store.coin += initial_balance;
```

Comment:
nit "=" instead of "+=" is cleaner, unless it's deliberate, where a comment is desired.